sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/controller/group/ThroughputGroupControllerBase.java

```diff
@@ -0,0 +1,238 @@
+
+    private void calculateGroupThroughput() {
+        double allocatedThroughput = Double.MAX_VALUE;
+        if (this.group.getTargetThroughputThreshold() != null) {
+            allocatedThroughput = Math.min(allocatedThroughput, this.maxContainerThroughput.get() * this.group.getTargetThroughputThreshold());
+        }
+
+        if (this.group.getTargetThroughput() != null) {
+            allocatedThroughput = Math.min(allocatedThroughput, this.group.getTargetThroughput());
+        }
+
+        this.groupThroughput.set(allocatedThroughput);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> Mono<T> init() {
+        return this.resolveRequestController()
+            .doOnSuccess(dummy -> {
+                scheduler.schedule(() -> this.throughputUsageCycleRenewTask(this.cancellationTokenSource.getToken()).subscribe());
+            })
+            .thenReturn((T)this);
+    }
+
+    private Flux<Void> throughputUsageCycleRenewTask(CancellationToken cancellationToken) {
+        checkNotNull(cancellationToken, "Cancellation token can not be null");
+        return Mono.delay(DEFAULT_THROUGHPUT_USAGE_RESET_DURATION)
+            .flatMap(t -> this.resolveRequestController())
+            .flatMap(requestController -> requestController.renewThroughputUsageCycle(this.groupThroughput.get()))
+            .onErrorResume(throwable -> {
+                logger.warn("Reset throughput usage failed with reason %s", throwable);
+                return Mono.empty();
+            }).repeat(() -> !cancellationToken.isCancellationRequested());
+    }
+
+    private Mono<IThroughputRequestController> createAndInitializeRequestController() {
+        IThroughputRequestController requestController;
+        if (this.connectionMode == ConnectionMode.DIRECT) {
+            requestController = new PkRangesThroughputRequestController(
+                this.globalEndpointManager,
+                this.partitionKeyRangeCache,
+                this.targetContainerRid,
+                this.groupThroughput.get());
+
+        } else if (this.connectionMode == ConnectionMode.GATEWAY) {
+            requestController = new GlobalThroughputRequestController(this.globalEndpointManager, this.groupThroughput.get());
+        } else {
+            throw new IllegalArgumentException(String.format("Connection mode %s is not supported"));
+        }
+
+        return requestController.init();
+    }
+
+    public boolean isUseByDefault() {
+        return this.group.isUseByDefault();
+    }
+
+    public void onContainerMaxThroughputRefresh(int maxContainerThroughput) {
+        if (this.maxContainerThroughput.getAndSet(maxContainerThroughput) != maxContainerThroughput) {
+            this.calculateGroupThroughput();
+        }
+    }
+
+    @Override
+    public Mono<Void> close() {
+        this.cancellationTokenSource.cancel();
+        return this.resolveRequestController()
+            .flatMap(requestController -> requestController.close());
```

Comment:
Same comments on this `close()` as above.

Reply:
Ack, will update in the next iteration.

Reply:
As discussed offline, will keep init and close as Mono for now.